src/pybind/mgr/volumes/fs/misc_util.py

```diff
@@ -0,0 +1,215 @@
+        being used.
+        '''
+        if self.op == 'copy':
+            return 'copied'
+        elif self.op == 'purge':
+            return 'purged'
+        else:
+            log.critical('Class Stats is being used for an operation it '
+                         'doesn\'t recognize')
+
+    def get_size_stat(self, human=True):
+        if human:
+            x, y = get_size_h(self.size_h), get_size_h(self.size_t)
+        else:
+            x, y = self.size_h, self.size_t
+
+        return f'{x}/{y}'
+
+    def get_rfiles_stat(self, human=True):
+        if human:
+            x, y = get_num_h(self.rfiles_h), get_num_h(self.rfiles_t)
+        else:
+            x, y = self.rfiles_h, self.rfiles_t
+
+        return f'{x}/{y}'
+
+    def get_dirs_stat(self, human=True):
+        if human:
+            x, y = get_num_h(self.dirs_h), get_num_h(self.dirs_t)
+        else:
+            x, y = self.dirs_h, self.dirs_t
+
+        return f'{x}/{y}'
+
+    def get_slinks_stat(self, human=True):
+        if human:
+            x, y = get_num_h(self.slinks_h), get_num_h(self.slinks_t)
+        else:
+            x, y = self.slinks_h, self.slinks_t
+
+        return f'{x}/{y}'
+
+    def get_progress_report(self, human=True):
+        return {
+            'percentage completed': self.percent,
+            f'amount {self.oped_str}': self.get_size_stat(human=human),
+            f'regfiles {self.oped_str}': self.get_rfiles_stat(human=human),
+            f'dirs {self.oped_str}': self.get_dirs_stat(human=human),
+            f'symlinks {self.oped_str}': self.get_slinks_stat(human=human)
+        }
+
+    def get_progress_report_str(self):
+        progress_report_str = ''
+        for k, v in self.get_progress_report().items():
+            progress_report_str += f'{k} = {v}\n'
+
+        # remove extra new line character at end, since print and log.xxx()
+        # methods automatically add a newline at the end.
+        return progress_report_str[:-1]
+
+    def log_handled_amount(self):
+        log.info('Following are statistics for amount handled vs. total '
+                 'amount -')
+        for k, v in self.get_progress_report().items():
+            log.info(f'{k} = {v}')
+
+    def log_total_amount(self):
+        log.info('Following are statistics for source path -')
+        log.info(f'    total size = {get_size_h(self.size_t)}')
+        log.info(f'    total number of regular files = {get_num_h(self.rfiles_t)}')
+        log.info(f'    total number of directories = {get_num_h(self.dirs_t)}')
+        log.info(f'    total number of symbolic links = {get_num_h(self.slinks_t)}')
+
+
+def get_statistics_for_path(given_path, fsh, op, should_cancel):
+    '''
+    Get statistics like of total number of files and total size of these the
+    files at the given path.
+
+    This method was originally written to provide a progress report to user.
+    '''
+    def traverse_tree(given_path, stats):
```

Comment:
Hey @rishabh-d-dave - I expect this routine to be done away in your latest revision. There is absolutely no reason to have this around.

Reply:
I've gotten this done only partially in last few days since few tests too problematic. A couple of them are exposing a bug and one of them fails sometimes perhaps due to less RAM on my machine. So, I've not pushed it yet.

Since these tests have been figured out by me now, I intend to finish and update this PR. I'll ping you when this is done.

Reply:
Since rstats can be relatively inaccurate, it will make progress bars unstable. Therefore, I made writing tests and fixing bugs a higher priority. Now that this is done, rstats are the highest priority.

Reply:
Marking resolved, this has been implemented.